common/membership/hashring.go

```diff
@@ -296,16 +279,20 @@ func (r *ring) refreshRingWorker() {
 	refreshTicker := r.timeSource.NewTicker(defaultRefreshInterval)
 	defer refreshTicker.Stop()
 
+	callRefresh := func() {
+		if err := r.refresh(); err != nil {
+			r.logger.Error("failed to refresh ring", tag.Error(err))
+		}
+	}
+
 	for {
 		select {
 		case <-r.shutdownCh:
 			return
-		case <-r.refreshChan: // local signal or signal from provider
-			if err := r.refresh(); err != nil {
-				r.logger.Error("failed to refresh ring", tag.Error(err))
-			}
-		case <-refreshTicker.Chan(): // periodically force refreshing membership
-			r.signalSelf()
+		case <-r.debounce.Chan():
+			callRefresh()
+		case <-refreshTicker.Chan():
+			callRefresh()
```

Comment:
the earlier-code equivalent would probably be to `r.debounce.Handler()` inside the refresh ticker, so it's debounced with updates / we don't refresh 2x quickly.

but this seems fine either way tbh.  and in some ways doing it separately is more bug-tolerant since it ensures at least _some_ updates occur.
not requesting a change, just pointing out that it _was_ changed, in case that wasn't intended.